mos_tests/nova/nova_test.py

```diff
@@ -550,3 +502,45 @@ def test_live_migration_of_v_ms_with_data_on_root_and_ephemeral_disk(self):
         self.assertEqual(root_data, r_data, "Data on root disk is changed")
         self.assertEqual(ephem_data, ep_data, "Data on ephemeral disk is "
                                               "changed")
+
+    def live_migration(self, instance, ip_to_ping, timeout=20):
+        hypervisors = {h.hypervisor_hostname: h for h in
+                       self.nova.hypervisors.list()}
+        old_hyper = getattr(instance, "OS-EXT-SRV-ATTR:hypervisor_hostname")
+        new_hyper = [h for h in hypervisors.keys() if h != old_hyper][0]
+        # Start ping of the vm in background
+        ping = subprocess.Popen(["/bin/ping", "-c20", "-i1", ip_to_ping],
+                                stdout=subprocess.PIPE)
+        # Then run the migration
+        self.nova.servers.live_migrate(instance, new_hyper,
+                                       block_migration=True,
+                                       disk_over_commit=False)
+        # Check that migration is over, usually it takes about 10-15 seconds
+        end_time = time() + 60 * timeout
+        while getattr(instance, "OS-EXT-SRV-ATTR:hypervisor_hostname") != \
```

Comment:
Inherited from legacy code, have no time to change and test other solutions today.